controller/src/main/java/io/pravega/controller/fault/ControllerClusterListener.java

```diff
@@ -117,6 +125,16 @@ protected void startUp() throws Exception {
                 eventProcessorsOpt.get().handleOrphanedReaders(processes);
             }
 
+            if (txnSweeperOpt.isPresent()) {
+                // Await initialization of transactionTasksOpt.
+                log.info("Awaiting StreamTransactionMetadataTasks to get ready");
+                txnSweeperOpt.get().awaitInitialization();
```

Comment:
Here we let the Interrupted exception float up to the caller of this `startUp` method from `AbstractIdleService`'s startAsync method, with the understanding that AbstractIdleService shall handle Interrupted exception appropriately by failing the service. Could we do anything better here?